identify-layer-features/src/main/java/com/esri/arcgismaps/sample/identifylayerfeatures/components/MapViewModel.kt

```diff
@@ -0,0 +1,145 @@
+
+import android.app.Application
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.data.ServiceFeatureTable
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISMapImageLayer
+import com.arcgismaps.mapping.layers.FeatureLayer.Companion.createWithFeatureTable
+import com.arcgismaps.mapping.view.IdentifyLayerResult
+import com.esri.arcgismaps.sample.identifylayerfeatures.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class MapViewModel(
+    private val application: Application,
+    private val sampleCoroutineScope: CoroutineScope
+) : AndroidViewModel(application) {
+    // set the MapView mutable stateflow
+    val mapViewState = MutableStateFlow(MapViewState())
+
+    // create a ViewModel to handle dialog interactions
+    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    // string text to display the identify layer results
+    val bottomTextBanner = mutableStateOf(String())
+
+    init {
+        // create a feature layer of damaged property data
+        val featureTable = ServiceFeatureTable(application.getString(R.string.damage_assessment))
+        val featureLayer = createWithFeatureTable(featureTable)
+
+        // create a layer with world cities data
+        val mapImageLayer = ArcGISMapImageLayer(application.getString(R.string.world_cities))
+        sampleCoroutineScope.launch {
+            mapImageLayer.load().onSuccess {
+                mapImageLayer.apply {
+                    subLayerContents.value[1].isVisible = false
+                    subLayerContents.value[2].isVisible = false
+                }
+            }.onFailure { error ->
+                // show the message dialog and pass the error message to be displayed in the dialog
+                messageDialogVM.showMessageDialog(error.message.toString(), error.cause.toString())
+            }
+        }
+
+        // create a topographic map
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic).apply {
+            // add world cities layer
+            operationalLayers.add(mapImageLayer)
+            // add damaged property data
+            operationalLayers.add(featureLayer)
+        }
+        // assign the map to the map view
+        mapViewState.value.arcGISMap = map
+    }
+
+    /**
+     * Identify the feature layer results and display the resulting information
+     */
+    fun handleIdentifyResult(result: Result<List<IdentifyLayerResult>>) {
+        sampleCoroutineScope.launch {
+            result.onSuccess { identifyResultList ->
+                bottomTextBanner.value = ""
```

Comment:
Don't think this is needed here, the `bottomTextBanner` should be set if `totalCount > 0`, else, set the text to be "Number of elements found: N/A"
```suggestion
```